core/src/main/scala/besom/internal/Context.scala

```diff
@@ -58,7 +58,7 @@ trait Context extends TaskTracker:
   /** This is called by codegened custom resource & stack reference constructor functions, get functions and rehydrated resources decoders.
     */
   private[besom] def readOrRegisterResource[R <: Resource: ResourceDecoder, A: ArgsEncoder](
-    typ: ResourceType,
+    typ: Input[ResourceType],
```

Comment:
is this change acceptable @lbialy and do we want to do the same with the name below

Reply:
I've reverted this change, because I've reduced the scope of this PR